lib/spack/spack/schema/gitlab_ci.py

```diff
@@ -80,6 +89,7 @@
                 ],
             },
         },
+        "match_behavior": {"type": "string", "enum": ["first", "merge"]},
```

Comment:
One request: can you add a default here as well? There is an example of this in `schema/env.py` for `concretization` - in this case it would look like:

```
"match_behavior": {
    "type": "string",
    "enum": ["first", "merge"],
    "default": "first",
},
```

I see you effectively set a default in `ci.py`, but IMO it makes things clear when we set those defaults in the schema as well (I don't think any other changes are needed: e.g. you can keep `gitlab_ci.get("match_behavior", "first")`).

Other than that, this looks good.

Reply:
Done.

Is there a way to avoid duplicating the default value in the code? I tried using `gitlab_ci["match_behavior"]` but got `KeyErrors` when the entry wasn't given in the YAML. Am I missing something or is this a limitation in Spack's configuration handling?